### PR TITLE
BorderControl: Promote to stable

### PR DESCRIPTION
**Pull request by DaniGuardiola**
## What?

Part of https://github.com/WordPress/gutenberg/issues/59418

Promote `BorderControl` to a non-experimental component.

## How?

- [x] Export it from components without the `__experimental` prefix.
- [x] Switch subcomponents to dot notation if applicable.
- [x] Keep the old `__experimental` export for backwards compatibility.
- [x] Remove experimental callout from readme.
- [x] Change all imports of the old `__experimental` in GB and components to the one without the prefix (including in Storybook stories).
- [x] Update the docs (readme, JSDocs, Storybook) to refer to the new unprefixed component.
- [x] Add the component storybook `id` (get it from the storybook URL) to the `PREVIOUSLY_EXPERIMENTAL_COMPONENTS` const array in `manager-head.html` so that old experimental story paths are redirected to the new one.
- [x] Add a changelog for both the promotion (enhancement) and the deprecation.

## For discussion

There are two unstable/experimental props that need to be discussed @WordPress/gutenberg-components @WordPress/gutenberg-core 

- `__experimentalIsRenderedInSidebar`: comes from the `ColorPalette` component that's used internally. Description: "Whether this is rendered in the sidebar". I guess this needs to be stabilized in the `ColorPalette` component itself, which is already stable. This prop seems to be used in many places with different values, so my hunch is that we'll want to make it stable.
- `__unstablePopoverProps`: exclusively used in `BorderBoxControl`, described as "An internal prop used to control the visibility of the dropdown". It forwards props to `Popover` [through `Dropdown`](https://github.com/WordPress/gutenberg/blob/bc7ba3afa3e7ead95dcbf55fb653c98e5ca50913/packages/components/src/border-control/border-control-dropdown/component.tsx#L249). I'm not sure what we'll want to do here. `Popover` is stable but some of its props are unstable.

Are any of these safe to stabilize? Separately, should the stabilization of these props block stabilization of the component? Let me know what you think.

Edit: tracked in https://github.com/WordPress/gutenberg/issues/65581

### CODE REVIEW

File: packages/components/src/index.ts
Comment (youknowriad) [inline packages/components/src/index.ts:45]: Should we add a `deprecated` call when using this one?

Comment (mirka) [reply packages/components/src/index.ts:45]: [Context](https://github.com/WordPress/gutenberg/pull/60913/files#r1738930767)

Comment (youknowriad) [reply packages/components/src/index.ts:45]: What's the reasoning for this specific components? This and the AlignmentMatrix one?

Comment (mirka) [reply packages/components/src/index.ts:45]: There is no strong reason for a consumer to switch over, and no immediate need for the library to remove the back-compat code (which is just an alias).

Comment (youknowriad) [reply packages/components/src/index.ts:45]: > no immediate need for the library to remove the back-compat code (which is just an alias).

This is true but in my perspective, if we don't deprecate, this will never change. But if we do deprecate now, we'll be in a better position one/two years from now to actually remove the component if we wanted to.

This is why I think if we consider something deprecated, we should always add the deprecated call. It opens more doors for us in the future without any downside for the present. 

The only cases where I won't do that is if it's something that is too impactful and that is likely to trigger warnings for all users... I would add a step there: soft deprecate and communicate and add a deprecation warning a bit later. It doesn't seem to be the case for me for these components.

Comment (mirka) [reply packages/components/src/index.ts:45]: This type of pros/cons weighing is already considered in the guidelines I proposed in #61099. There is a very tangible cost we shift to the consumers when we log a deprecated call. They need to hunt it down and address it, even if they get zero functional benefit in doing so. Whereas, indefinitely keeping a couple lines of back-compat code in the library is virtually free.

And when it does start to become not free for the library in aggregate, it's not too late to log a warning then. Maybe that is what you're proposing though?

Comment (DaniGuardiola) [reply packages/components/src/index.ts:43]: That might be the way to go. We can always add deprecation warnings in a couple of WP versions after all components have been stabilized, though I'm not sure we'll ever be able to remove the experimental exports for certain components.

Comment (youknowriad) [reply packages/components/src/index.ts:45]: > And when it does start to become not free for the library in aggregate, it's not too late to log a warning then. Maybe that is what you're proposing though?

I'm saying that it's too late at that point to add a warning because it will cause the "removal" to be delayed by a couple years at least (people to notice the new warnings and move away from it).

On the opposite site, if you don't show the warnings, there's no incentive for people to move out. so why it's deprecated in the first place?

> There is a very tangible cost we shift to the consumers when we log a deprecated call.

The cost is very small in most cases, this is what I'm arguing about. Very few people outside Core are using these two components, so the cost is small compare to the pros of incentivizing these users to move away from these deprecated APIs. At what point the cost becomes too big that it requires an initial period of communication (make/core post) before deprecation is the right question to ask IMO. I think this would be rare and only for very largely used APIs.

Comment (mirka) [reply packages/components/src/index.ts:45]: > I'm saying that it's too late at that point to add a warning because it will cause the "removal" to be delayed by a couple years at least (people to notice the new warnings and move away from it).

The thing is that we've been maintaining components for three years or so now, and we've never experienced this level of urgency to hard remove deprecated code. And for things that we do think would be better to hard remove quickly, we do in fact log deprecation warnings as soon as possible.

> On the opposite site, if you don't show the warnings, there's no incentive for people to move out. so why it's deprecated in the first place?

This is also why we wanted to define different levels of deprecation. Passive deprecations are exactly for cases where we have no reason for people to move out. It's merely an administrative change for the library. It adds clarity/simplicity for new consumers, but existing usages don't gain anything by switching over.

> The cost is very small in most cases, this is what I'm arguing about. Very few people outside Core are using these two components, so the cost is small compare to the pros of incentivizing these users to move away from these deprecated APIs.

This is about a subjective perception of cost, so it's hard to weigh objectively. At least from the usage checks we've been doing for components, [AlignmentMatrixControl](https://wpdirectory.net/search/01J85APV8KKMY2JS9NSPZK85H7) and [BorderControl](https://wpdirectory.net/search/01J85AGHXDWZS02XAYS8N2GND6) are not low-usage, relatively speaking. And we've been getting feedback from extenders that it's definitely a hassle to deal with deprecation warnings, especially in a way that works across WP versions. To me, keeping two lines of back-compat code is cheaper than the cost on the ecosystem to switch over for no tangible benefit. And whenever we start to feel the cost analysis flipping, it likely won't be too late.

I actually think it's more likely that we'll reach a point where these (stable) components need to be deprecated entirely and/or superseded by a v2, sooner than we'll reach a point where the back-compat code is too costly to maintain.

Comment (youknowriad) [reply packages/components/src/index.ts:45]: > This is also why we wanted to define different levels of deprecation. Passive deprecations are exactly for cases where we have no reason for people to move out. It's merely an administrative change for the library. It adds clarity/simplicity for new consumers, but existing usages don't gain anything by switching over.

I'm not sure I agree that keeping deprecated APIs forever is a good idea personally. Also, there's no urgency today but there might be tomorrow. So having the possibility to remove a deprecated API tomorrow is for me a good thing to have. 

Say, two years have passed and for some random reason, I don't want to keep these around? I would be very happy to know that we had a message warning developers that these are deprecated for two years now. That message is costless IMO, neither for us, nor for the very few people that actually use that component. 

Also, I don't like that we treat the "components" package as separate from all the other APIs in Gutenberg. If we have a strategy, we should have a consistent one.

Comment (youknowriad) [reply packages/components/src/index.ts:45]: While I stand by my reasoning. I don't mind disagreeing and committing to any approach here.

I think the important piece of feedback for me here is the latter 

> Also, I don't like that we treat the "components" package as separate from all the other APIs in Gutenberg. If we have a strategy, we should have a consistent one.

I think we should stop treating components as a special package and whatever we do for that package, we should do for the rest.

**Comment by DaniGuardiola** [comment]
@youknowriad thanks for chiming in! I would love your thoughts on the props mentioned in the "For discussion" section of the description, if you have the time :) cc @WordPress/gutenberg-components too

**Comment by youknowriad** [comment]
> __experimentalIsRenderedInSidebar

What does it actually do? It might be too app specific to stabilize.

> __unstablePopoverProps

I'm guessing this is mostly used to control the position of the dropdown menu. It feels like a use-case that is needed but I'm not sure what's the best API here.

Having said that, we do have stable components with unstable props already. Ideally, these could be "private props" maybe until we figure out what to do with them but I also don't mind shipping as they already exist.

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for putting this together @DaniGuardiola 👍 

> `__experimentalIsRenderedInSidebar`: ... I guess this needs to be stabilized in the ColorPalette component itself, which is already stable. This prop seems to be used in many places with different values, so my hunch is that we'll want to make it stable.

+1 for stabilizing this in the ColorPalette via a separate follow-up.


> `__unstablePopoverProps`: ... It forwards props to Popover through Dropdown. I'm not sure what we'll want to do here. Popover is stable but some of its props are unstable.

I'm not sure about stabilizing the `__unstablePopoverProps` on the `BorderBoxControl` either. 

It was only intended to be an internal means of passing popover props from the parent `BorderBoxControl` to the inner `BorderControl` components. 

There some history in https://github.com/WordPress/gutenberg/pull/40740#discussion_r869585764. One option there was to try using the component context system for the components to "talk" to each other. I'm not sure what ended up happening with that approach. It might allow us to avoid stabilizing `__unstablePopoverProps` and instead remove it entirely.

> \_\_experimentalIsRenderedInSidebar
> ...
> What does it actually do? It might be too app specific to stabilize.

My understanding is that the prop [adjusts positioning of the popovers](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/color-palette/index.tsx?plain=1#L154-L165) to account for the sidebar which does seem to app specific too stabilize.



> Ideally, these could be "private props" maybe until we figure out what to do with them

Agreed.

---

I've also given this a bit of a test. From a functional perspective it's looking good ✨ 

✅ Smoke testing in editors didn't reveal any issues
✅ Component unit tests are all passing
✅ Storybook examples for BorderControl function as expected
✅ Experimental BorderControl URL in Storybook is redirected to stable version
✅ Didn't find any further references to `__experimentalBorderControl` than the export kept for BC (pending discussion)

**Comment by ciampo** [comment]
> Are any of these safe to stabilize? Separately, should the stabilization of these props block stabilization of the component? Let me know what you think.

Conceptually, I'd love to tackle the props before stabilizing the component.

But:

- The component is already expected to be "stable" (even with the experimental prefix) because it's been part of Core releases;
- Working on individual props would potentially delay the stabilization of the component by a long time.

So, practically speaking, we may want to stabilize this component and then, separately, work on the individual props. I like the idea of making them private props for the time being.

> * `__experimentalIsRenderedInSidebar`: comes from the `ColorPalette` component that's used internally. Description: "Whether this is rendered in the sidebar". I guess this needs to be stabilized in the `ColorPalette` component itself, which is already stable. This prop seems to be used in many places with different values, so my hunch is that we'll want to make it stable.

Ideally, we should get rid of this prop. The concept of the "sidebar" only makes sense in the Gutenberg app, and should be foreign to the components in the `@wordpress/components` package.

IMO, we should:
- look at what exactly this prop causes;
- find alternative ways to achieve the same result, without introducing app-specific concepts;
- remove/deprecate the prop.

> * `__unstablePopoverProps`: exclusively used in `BorderBoxControl`, described as "An internal prop used to control the visibility of the dropdown". It forwards props to `Popover` [through `Dropdown`](https://github.com/WordPress/gutenberg/blob/bc7ba3afa3e7ead95dcbf55fb653c98e5ca50913/packages/components/src/border-control/border-control-dropdown/component.tsx#L249). I'm not sure what we'll want to do here. `Popover` is stable but some of its props are unstable.

Here we should also look at what this prop is used for. If, like Aaron says, it's only used to communicate between different components in the `@wordpress/components` package, then we could definitely use the context system and deprecate the prop.

**Comment by tyxla** [comment]
> Conceptually, I'd love to tackle the props before stabilizing the component.
> 
> But:
> 
> * The component is already expected to be "stable" (even with the experimental prefix) because it's been part of Core releases;
> * Working on individual props would potentially delay the stabilization of the component by a long time.
> 
> So, practically speaking, we may want to stabilize this component and then, separately, work on the individual props. I like the idea of making them private props for the time being.
> 
> >

I think this will be the case for many components, but if we wait until each component is perfect, some might never get stabilized. What you described on the practical side seems like the best option for me in the long term, as long as we document everything we intend to do in the future.

File: packages/components/src/index.ts
Comment (DaniGuardiola) [reply packages/components/src/index.ts:45]: Let's continue this discussion elsewhere - if a decision is made to apply deprecation warnings we can do so in all affected components very easily at a later moment.

**Comment by DaniGuardiola** [comment]
I created https://github.com/WordPress/gutenberg/issues/65581 to track those. Can I get an approval here to merge?

**Comment by aaronrobertshaw** [comment]
> What you described on the practical side seems like the best option for me in the long term, as long as we document everything we intend to do in the future.

+1

> I created https://github.com/WordPress/gutenberg/issues/65581 to track those. Can I get an approval here to merge?

Thanks for creating the issue.

I've given this another smoke test and it is still working well for me. In terms of overall approach, I'm happy to defer to @ciampo and @tyxla here and it appears they're happy. So I'm happy to approve this one ✅

**Comment by DaniGuardiola** [comment]
Btw, anyone have any idea why those tests are failing? It's happening for E2E and React Native tests across many of my PRs, and I'm clueless as to why. I also posted in the Slack: https://wordpress.slack.com/archives/C02QB2JS7/p1727167122935539

**Comment by aaronrobertshaw** [comment]
> Btw, anyone have any idea why those tests are failing? It's happening for E2E and React Native tests across many of my PRs, and I'm clueless as to why. I also posted in the Slack:

I believe the failures were related to a core commit: https://github.com/WordPress/wordpress-develop/commit/805b9333f2de80c3db981e1a2a5755045d9e7782

There's more context in the [comment](https://wordpress.slack.com/archives/C02QB2JS7/p1727152145938689) right above yours in that same Slack channel.

File: packages/components/src/border-control/border-control/README.md
Comment (tyxla) [inline packages/components/src/border-control/border-control/README.md]: Shouldn't this be `true` actually?

https://github.com/WordPress/gutenberg/blob/72436e1f365a5c099285c92e750bcb3adbbb3a4c/packages/components/src/border-control/border-control/hook.ts#L33

Comment (tyxla) [inline packages/components/src/border-control/border-control/README.md]: If the above is correct and `enableAlpha` defaults to `true` intentionally, shouldn't we invert the `WithAlphaEnabled` story to demonstrate a "WithAlphaDisabled" behavior instead?

File: packages/components/src/border-control/stories/index.story.tsx
Comment (tyxla) [inline packages/components/src/border-control/stories/index.story.tsx:19]: Similar to #65586 should we consider fixing the default bool prop value in Storybook:


https://github.com/user-attachments/assets/b2af5552-c9cd-4853-9310-dfb75f39fbfb

by adding the `true` to the story for all boolean props that default to `true` in the hook:

```diff
diff --git a/packages/components/src/border-control/stories/index.story.tsx b/packages/components/src/border-control/stories/index.story.tsx
index 19938db5d0e..1bcb22c50d6 100644
--- a/packages/components/src/border-control/stories/index.story.tsx
+++ b/packages/components/src/border-control/stories/index.story.tsx
@@ -93,6 +93,9 @@ export const Default = Template.bind( {} );
 Default.args = {
        colors,
        label: 'Border',
+       enableAlpha: true,
+       enableStyle: true,
+       shouldSanitizeBorder: true,
 };

 /**
```

File: packages/components/src/border-control/border-control/README.md
Comment (tyxla) [inline packages/components/src/border-control/border-control/README.md]: Thinking about the API, I wonder if this made sense to be a boolean prop. Just like we have a `label`, it might have made more sense to have a prop that allows controlling that dropdown header label. And if it's specified, we would automatically display the close button. 

@ciampo @mirka WDYT?

Maybe it's a good time to change this if y'all agree. The prop is used only internally in the package.

**Review by tyxla — COMMENTED** [review]
Thanks for working on it @DaniGuardiola. Looks good overall!

I have a few comments / questions, LMK what you think.

File: packages/components/src/border-control/border-control/README.md
Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: Fixed. I'd like to try generating these automatically in the future so that they're kept in sync with the actual types and the JSDocs, including defaults.

Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: Makes sense, done.

File: packages/components/src/border-control/stories/index.story.tsx
Comment (DaniGuardiola) [reply packages/components/src/border-control/stories/index.story.tsx:19]: Yep, makes sense. Done.

File: packages/components/src/border-control/border-control/README.md
Comment (tyxla) [reply packages/components/src/border-control/border-control/README.md]: Absolutely. It was a surprise to find this there in the first place.

File: packages/components/src/border-control/stories/index.story.tsx
Comment (tyxla) [reply packages/components/src/border-control/stories/index.story.tsx:19]: For the protocol, this is another thing that should ideally happen automatically. Right now, it feels like a Storybook glitch.

File: packages/components/src/border-control/border-control/README.md
Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: My thoughts:

- If we want to allow customization, we'll need a `colorHeaderLabel` sort of prop.
- I'm not sure I can see a use case for a custom label here though. This is a part of a border control, so while I can see a custom label at the top-level (e.g. "\<custom block name or whatever> border") I struggle to find a similar case for this sub-setting.
- Even if we allow customization, it might make sense to have a default. In that case, a prop like this would still make sense.
- A good middle ground could be to keep this `false` by default, but make it `true` by default if the `colorHeaderLabel` prop is set. That way we reduce the chance of accidentally not showing the label even though a custom one has been set.

My conclusion: based on 1. that I don't currently see a use case for a custom label and 2. that even if so, a prop like this would still make sense (with the sensible default heuristic described), I feel like the best path forward is to leave this as-is and worry about it again if/when the use-case for custom label arises, if that ever happens.

Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: Alternative suggestion: just show the header at all times and remove this setting.

File: packages/components/src/border-control/stories/index.story.tsx
Comment (DaniGuardiola) [reply packages/components/src/border-control/stories/index.story.tsx:19]: I remember looking into this years ago. There was something about Storybook picking up the default automatically in controls. Don't really remember the details though.

File: packages/components/src/border-control/border-control/README.md
Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: On second thought, is this story really that helpful? Maybe it is in the underlying component (`ColorPalette`, which btw doesn't have it), but here it seems pretty useless when the control is right there. We might be able to get away with just removing it.

Comment (tyxla) [reply packages/components/src/border-control/border-control/README.md]: Could be useful if we want to highlight this particular use case, but I agree that the story isn't as helpful as many others we have. I'm not opposed to removing it 👍

File: packages/components/src/border-control/border-control/README.md
Comment (tyxla) [reply packages/components/src/border-control/border-control/README.md]: Keeping the default behavior sounds like a good alternative to me. But isn't the default behavior to NOT show the header ATM?

Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: > Keeping the default behavior sounds like a good alternative to me. But isn't the default behavior to NOT show the header ATM?

I'm a bit lost, what do you mean here? Just removing the option altogether?

File: packages/components/src/border-control/border-control/README.md
Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: Yeah I think I'm just going to remove it. Doesn't seem useful vs just using controls.

File: packages/components/src/border-control/border-control/README.md
Comment (tyxla) [reply packages/components/src/border-control/border-control/README.md]: Yes. If we never show that label, removing the option does make sense.

Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: That'd be one more option to take here. Doing a quick lookup, it doesn't seem like it's ever been used at all.

Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: Origin:

> I've just added a prop to toggle its display for now. This is included in the Storybook example controls so it can be evaluated.

Source: https://github.com/WordPress/gutenberg/pull/37769#issuecomment-1074809008

Looping @jasmussen and @WordPress/gutenberg-design for feedback, but it seems like no heading became the de-facto design and we should be able to remove it altogether.

Comment (jameskoster) [reply packages/components/src/border-control/border-control/README.md]: I don't see any benefit to making this optional—it only opens the door to inconsistencies.

Whether to display the header/close button or not; we're already a bit inconsistent about this detail across the various popovers in Gutenberg, so I don't have a strong opinion. But generally I'd agree that maintaining the current design makes the most sense for now. 

We can revisit all popovers holistically once there is consensus in https://github.com/WordPress/gutenberg/issues/63674.

Comment (tyxla) [reply packages/components/src/border-control/border-control/README.md]: Thanks for verifying @jameskoster! 

If the current design is to always hide it, and we never use it, I'd just suggest removing the prop altogether.

Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: Not just that, we should remove the heading itself too. Will do very soon.

Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: Done in https://github.com/WordPress/gutenberg/pull/65475/commits/d357efda081be89758b69f2eefc0f27a68127670

**Comment by DaniGuardiola** [comment]
I believe everything has been addressed, should be ready for merge if nothing else comes up :)

File: packages/components/src/border-control/border-control/README.md
Comment (ciampo) [reply packages/components/src/border-control/border-control/README.md]: > I'd like to try generating these automatically in the future so that they're kept in sync with the actual types and the JSDocs, including defaults.

Yup, I believe that's part of Lena's future planned work

File: packages/components/src/border-control/stories/index.story.tsx
Comment (ciampo) [reply packages/components/src/border-control/stories/index.story.tsx:19]: Definitely a storybook bug

File: packages/components/src/border-control/border-control/README.md
Comment (ciampo) [reply packages/components/src/border-control/border-control/README.md]: Sorry if I'm late to the party, but by removing the type def, we may introduce breaking changes to folks specifying that prop (they may get TS errors). A "softer" alternative would be to keep the type def, adding the `@deprecated` JSDoc tag to deprecated it, and the `@ignore` JSDoc tag to exclude it from JSDocs snippets and Storybook.

cc @mirka to double-check this is a good approach.

**Comment by tyxla** [comment]
> I believe everything has been addressed, should be ready for merge if nothing else comes up :)

Unless I'm missing something, the only thing that remains is keeping the removed prop type as a deprecated / ignored one, to prevent TS errors, as @ciampo [recommended](https://github.com/WordPress/gutenberg/pull/65475#discussion_r1778414108).

File: packages/components/src/border-control/border-control/README.md
Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: What do you mean by "folks specifying that prop"? In Gutenberg it's not used anywhere like the prop name string literally doesn't exist anywhere else. Do you mean external consumers of the component?

Comment (ciampo) [reply packages/components/src/border-control/border-control/README.md]: Yes, I mean external consumers. Keeping the typedef around doesn't cost us much, and prevents folks from dealing with build errors.

We can always add JSDocs explaining that basically the prop is not used anymore, and maybe even a dev note about it if we think it's needed?

Comment (mirka) [reply packages/components/src/border-control/border-control/README.md]: Agreed with Marco's suggestion. And this is technically a breaking change for extenders, but I'm ok with silently removing the feature since it won't cause any functional harm.

Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/README.md]: Alright, will do. Gonna add a deprecation warning though, so that we can remove it at some point in the future.

**Comment by DaniGuardiola** [comment]
I added the prop type back with `@deprecated` and a deprecation warning. Let me know if I did it right as it's the first one I add. I set it as deprecated since v6.7 as it's the upcoming version WP afaik, and to be removed in v7.

File: packages/components/src/border-control/border-control/component.tsx
Comment (tyxla) [inline packages/components/src/border-control/border-control/component.tsx]: If we want to specify a "to be removed in" version, let's make it follow the same pattern as everywhere, since each "semver major + minor" WordPress version is considered major.

```suggestion
			version: '7.0',
```

**Review by tyxla — APPROVED** [review]
Looking good from my perspective 👍 

Thanks @DaniGuardiola 🚀

File: packages/components/src/border-control/border-control/component.tsx
Comment (mirka) [reply packages/components/src/border-control/border-control/component.tsx]: I don't quite understand this warning though. We already removed the functionality in this branch so the removal is effective immediately, not in 7.0.

Personally, I prefer not logging a warning at all, because there's really no action that a consumer could take, other than stop passing the prop. And there's no reason to stop passing the prop, hence our decision to keep the TS compiling for convenience.

If you're worried about one day wanting to remove the TS definition, I think we're free to remove that at any time without warning.

File: packages/components/src/border-control/types.ts
Comment (mirka) [inline packages/components/src/border-control/types.ts:88]: Like @ciampo [suggested](https://github.com/WordPress/gutenberg/pull/65475#discussion_r1778414108), we should add an `@ignore` to suppress this prop from the Storybook props table.

Comment (DaniGuardiola) [reply packages/components/src/border-control/types.ts:88]: Ah right, forgot about that, will do thanks!

File: packages/components/src/border-control/border-control/component.tsx
Comment (DaniGuardiola) [reply packages/components/src/border-control/border-control/component.tsx]: Makes sense, I removed the warning before merge 👍

File: packages/components/CHANGELOG.md
Comment (ciampo) [inline packages/components/CHANGELOG.md:1]: Just flagging that this PR seems to have removed unintentionally some CHANGELOG entries.

Comment (DaniGuardiola) [reply packages/components/CHANGELOG.md:1]: oof, bad automatic merge, let me put up a quick PR

Comment (DaniGuardiola) [reply packages/components/CHANGELOG.md:1]: https://github.com/WordPress/gutenberg/pull/65804

**Pull request merged**
